File: typescript/patterns-use-cases/src/schedulingtasks/utils/stubs.ts
export type StripeEvent = {
    type: string;
    created: number;
    data: {
        id: string;
        customer: string;
    };
}

export function sendReminderEmail(event: StripeEvent) {
    console.log(`Sending reminder email for event: ${event.data.id}`)
}

export function escalateToHuman(event: StripeEvent) {
    console.log(`Escalating to ${event.data.id} invoice to support team`)
}

File: go/patterns-use-cases/src/schedulingtasks/stubs.go
package main

import "log/slog"

type StripeEvent struct {
	ID      string `json:"id"`
	Type    string `json:"type"`
	Created int64  `json:"created"`
	Data    struct {
		ID       string `json:"id"`
		Customer string `json:"customer"`
	} `json:"data"`
}

func SendReminderEmail(event StripeEvent) error {
	slog.Info("Sending reminder email for event " + event.Data.ID)
	return nil
}

func EscalateToHuman(event StripeEvent) error {
	slog.Info("Escalating to human for event " + event.Data.ID)
	return nil
}
